Internal builds are failing to compile JavaScriptCoreTestTools_Sim

Smackteo opened the pull request:
#### d7a1213c6583d4a384d5ca05b0ab237e51f759ee
<pre>
Internal builds are failing to compile JavaScriptCoreTestTools_Sim
<a href="https://rdar.apple.com/163951890">rdar://163951890</a>

Reviewed by Elliott Williams and Jonathan Bedard.

This adds the -Wno-error=stack-exhausted workaround which supresses the failure message.

* Source/JavaScriptCore/Configurations/TestExecutable.xcconfig:

Canonical link: <a href="https://commits.webkit.org/302553@main">https://commits.webkit.org/302553@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/67068eacde506ff3a2daf459c543910137205059

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/129454 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/1711 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/40293 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/136831 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/80873 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/7945535e-5d11-427f-afee-59bb629e8118) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/1643 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/1587 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/98594 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/66483 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/d72242ba-6ef6-4556-bc48-2d34498fa4ad) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/132401 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/162/builds/1282 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/115950 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/79246 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/9aed4b6e-9c0d-4b9c-bfbf-5f5e4e0c3762) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/1202 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/34079 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/80108 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/121446 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/109665 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/34578 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/139306 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/127906 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/1501 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/1439 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/107119 "Found 4 new test failures: http/wpt/service-workers/service-worker-process-crashing-between-installed-and-activating.html imported/w3c/web-platform-tests/encoding/legacy-mb-japanese/shift_jis/sjis-encode-href-errors-han.html?14001-15000 imported/w3c/web-platform-tests/webrtc/RTCDTMFSender-ontonechange-long.https.html imported/w3c/web-platform-tests/webrtc/RTCPeerConnection-addIceCandidate-timing.https.html (failure)") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/1543 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/112291 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/106963 "Passed tests") | | 
| [  ~~🛠 🧪 merge~~](https://ews-build.webkit.org/#/builders/19/builds/27235 "The change is no longer eligible for processing.") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/1224 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/30811 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/54173 "Built successfully") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/20204 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/1572 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/64935 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/160920 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/1391 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/40140 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/1426 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/1494 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

emw-apple reviewed (changes requested):
Please provide some more context -- what's wrong, and why is this an acceptable change? We should not be turning off errors without a good reason.

emw-apple commented on Source/JavaScriptCore/Configurations/TestExecutable.xcconfig:
```diff
@@ -39,7 +39,8 @@ SKIP_INSTALL_YES = NO;
 
-OTHER_CFLAGS = $(inherited) -isystem icu;
+OTHER_CFLAGS = $(inherited) -isystem icu -Wno-error=stack-exhausted;
+// Remove -Wno-error=stack-exhausted after rdar://163951890 is fixed in all active toolchains
```
Typically the comment would come before the code line.

emw-apple reviewed (approved):
Thanks for the clarifying breadcrumbs. Could you reword the commit message before landing? "This adds the workaround to the failure." doesn't really make sense.

Smackteo committed:
Internal builds are failing to compile JavaScriptCoreTestTools_Sim
rdar://163951890

Reviewed by Elliott Williams and Jonathan Bedard.

This adds the -Wno-error=stack-exhausted workaround which supresses the failure message.

* Source/JavaScriptCore/Configurations/TestExecutable.xcconfig:

Canonical link: https://commits.webkit.org/302553@main
